packages/firestore/test/unit/local/web_storage_shared_client_state.test.ts

```diff
@@ -360,14 +368,16 @@ describe('WebStorageSharedClientState', () => {
           ).toLocalStorageJSON()
         );
 
+        await queue.drain();
```

Comment:
Would it make sense to just make this part of withUser() ?

Reply:
Yeah, I was thinking about that too. I pushed two versions of this (the first commit just adds it to withUser, the second commit changes the return type to return the state that we assert on). Let me know what you think.